apps/mobile/src/components/Gallery/GalleryTokenPreview.tsx

```diff
@@ -40,36 +48,50 @@ export function GalleryTokenPreview({ tokenRef, containerWidth }: GalleryTokenPr
     }
   }, [containerWidth, token.token]);
 
-  const { cache, addDimensionsToCache } = useGalleryTokenDimensionCache();
+  // We need this as a fallback in case the backend is missing dimensions
+  const [measuredDimensions, setMeasuredDimensions] = useState<Dimensions | null>(null);
 
   const screenDimensions = useWindowDimensions();
   const handleImageStateChange = useCallback(
     (imageState: ImageState) => {
       if (imageState.kind === 'loaded' && imageState.dimensions && tokenUrl) {
-        addDimensionsToCache(tokenUrl, imageState.dimensions);
+        setMeasuredDimensions(imageState.dimensions);
       }
     },
-    [addDimensionsToCache, tokenUrl]
+    [tokenUrl]
   );
 
   const resultDimensions = useMemo(() => {
     if (!tokenUrl) {
       return null;
     }
 
-    const cachedDimensions = cache.get(tokenUrl) ?? null;
-
-    if (cachedDimensions) {
-      const result = fitDimensionsToContainerContain({
+    const serverSourcedDimensions = token.token.media?.dimensions;
+    if (serverSourcedDimensions?.width && serverSourcedDimensions.height) {
+      return fitDimensionsToContainerContain({
         container: { width: containerWidth, height: screenDimensions.height / 2 },
-        source: cachedDimensions,
+        source: {
+          width: serverSourcedDimensions.width,
+          height: serverSourcedDimensions.height,
+        },
       });
+    }
 
-      return result;
+    if (measuredDimensions) {
+      return fitDimensionsToContainerContain({
+        container: { width: containerWidth, height: screenDimensions.height / 2 },
+        source: measuredDimensions,
+      });
     }
```

Comment:
In this part, we're just trying to use the server dimensions as a primary source of truth, and then fallback to the measured dimensions.